Weird compiler warnings/messages on Windows?

Looks as if there might be some issue with the `==` operator override for the `Feature` class.
```
warning C4834: discarding return value of function with 'nodiscard' attribute
  D:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\xutility(5482): note: see reference to function template instantiation 'bool Feature::==<_Ty>(const _T0 &)' being compiled
          with
          [
              _Ty=Feature,
              _T0=Feature
          ]
  D:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\xutility(5527): note: see reference to function template instantiation '_InIt std::_Find_unchecked<Feature*,_Ty>(_InIt,const _InIt,const _Ty &)' being compiled
          with
          [
              _InIt=Feature *,
              _Ty=Feature
          ]
  FeatureExtraction\src\core/feature_finding/FeatureFinding.h(118): note: see reference to function template instantiation '_InIt std::find<std::_Vector_iterator<std::_Vector_val<std::_Simple_types<_Ty>>>,_Ty2>(_InIt,const _InIt,const _Ty &)' being compiled
          with
          [
              _InIt=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<Feature>>>,
              _Ty=Feature,
              _Ty2=Feature
          ]
```

--- a/P8/DataConverters/ShapeletHistogramConverter/FeatureExtraction/src/types/Feature.h
+++ b/P8/DataConverters/ShapeletHistogramConverter/FeatureExtraction/src/types/Feature.h
@@ -15,7 +15,7 @@ struct Feature {
 
     bool operator==(const auto &rhs) {
         return (
-                attribute == rhs.attribute,
+                attribute == rhs.attribute &&
                 shapelet == rhs.shapelet
                 );
     }

